src/DssSpell.t.sol

```diff
@@ -865,17 +910,251 @@ contract DssSpellTest is DssSpellTestBase {
+    JarLike           immutable jar           = JarLike(          addr.addr("MCD_LITE_PSM_USDC_A_JAR"));
+    InputConduitLike  immutable inputConduit  = InputConduitLike( addr.addr("MCD_LITE_PSM_USDC_A_IN_CDT_JAR"));
+    GemAbstract       immutable gem           = GemAbstract(      addr.addr("USDC"));
+    address           immutable pip           =                   addr.addr("PIP_USDC");
+    CronSequencerLike immutable sequencer     = CronSequencerLike(addr.addr("CRON_SEQUENCER"));
+    LitePsmJobLike    immutable litePsmJob    = LitePsmJobLike(   addr.addr("CRON_LITE_PSM_JOB"));
+    uint256           constant  dstBuf        =  20_000_000 * WAD;
+    uint256           constant  dstWant       =  20_000_000 * WAD;
+    uint256           constant  srcKeep       = 200_000_000 * WAD;
+    uint256           constant  rushThreshold =  15_000_000 * WAD;
+    uint256           constant  gushThreshold =  30_000_000 * WAD;
+    uint256           constant  cutThreshold  =     300_000 * WAD;
+
+    function test_LITE_PSM_USDC_A_CronJob() public {
+        // ----- Pre-spell sanity checks -----
+
+        assertFalse(sequencer.hasJob(address(litePsmJob)));
+
+        // Sequencer matches
+        assertEq(litePsmJob.sequencer(),     address(sequencer), "invalid sequencer");
+        // LitePsm matches
+        assertEq(litePsmJob.litePsm(),       address(dstPsm),    "invalid litePsm");
+        // fill: Set threshold at 15M DAI
+        assertEq(litePsmJob.rushThreshold(), rushThreshold,      "invalid rush threshold");
+        // trim: Set threshold at 30M DAI
+        assertEq(litePsmJob.gushThreshold(), gushThreshold,      "invalid rush threshold");
+        // chug: Set threshold at 300k DAI
+        assertEq(litePsmJob.cutThreshold(),  cutThreshold,       "invalid rush threshold");
+
+        // ----- Execute spell -----
+
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        // ----- Post-spell sanity checks -----
+
+        assertTrue(sequencer.hasJob(address(litePsmJob)));
+
+        // ----- E2E tests -----
+
+        bytes32 master = sequencer.getMaster();
+
+        // Base state
+        {
+            (bool ok, ) = litePsmJob.workable(master);
+            assertFalse(ok, "invalid workable ok - no changes");
+        }
+
+        // Modify local `line` so it is not a limiting factor
+        GodMode.setWard(address(vat), address(this), 1);
+        vat.file(DST_ILK, "line", type(uint256).max);
+
+        // Allow the test contract to swap with no fees
+        GodMode.setWard(address(dstPsm), address(this), 1);
+        dstPsm.kiss(address(this));
+
+        // Approvals
+        gem.approve(address(dstPsm), type(uint256).max);
+        dai.approve(address(dstPsm), type(uint256).max);
+
+        uint256 snapshot = vm.snapshot();
+
+        // Sell gems to require a fill
+        {
+            uint256 wadOut = rushThreshold + (1 * WAD); // Must be > rushThreshold
```

Comment:
On the other note: I actually think that it's more important to check that `rush()` above 0 but below `rushThreshold` does not produce workable state. Otherwise, this job will be callable after every LitePSM exchange operation and will become extremely expensive. I don't think this was tested in the [integration test](https://github.com/makerdao/dss-cron/blob/2f9aba4ab16603a77b55333b8ebaf348f5fe7e47/src/tests/LitePsmJob-integration.t.sol) (please correct me if I'm wrong)

Nevertheless, I've checked that manually, but still suggest to add the tests for all 3 thresholds.

Reply:
Regarding your last point, it's kind of a pain to define how extensive the tests for onboarding new modules/jobs must be. In this case, the coverage in the job itself is supbar, however it can be pretty difficult to test everything from within the spell test suite.

Maybe we should just test what the spell is actually doing and shift that burden to those implementing the code being onboarded. This will be especially important as we onboard several endgame modules in a single spell.

I'll add coverage for such cases here, however I believe we need to discuss the strategy moving forward.

Reply:
> Maybe we should just test what the spell is actually doing and shift that burden to those implementing the code being onboarded

Yes and no. In the spells we normally only have to test that the modules are correctly configured, not the modules themselves. However, 
- Sometimes the easiest way to test correct configuration is to write a "happy path" end-to-end test
- Spells are the last place where all of the contracts (sometimes audited separately or not audited at all, deployed by third parties) come together and we have a responsibility to spot common sense issues, missing audits, reviews or even test coverage (although this one is not our direct responsibility). When anything crucial is missing I believe we have to make it public or even delay the onboarding

For this specific case, I already checked that there is no problem, my _suggestion_ to add coverage was just for the sanity and to reduce work for the other reviewer / next onboardings. Feel free to skip it.

Reply:
Created this [issue](https://github.com/makerdao/spells-mainnet/issues/418) so we can continue this discussion afterwards.